### PR TITLE
Switch to .net8.0 preview

### PR DESCRIPTION
**Pull request by marcin-dardzinski**
We have a very nice naming conflict with `System.TimeProvider` that will need addressing

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -95,7 +95,8 @@
     <PackageReference Update="Microsoft.Extensions.Logging.Debug" Version="$(ExtensionsVersion)" />
     <PackageReference Update="Microsoft.Extensions.Options" Version="$(ExtensionsVersion)" />
     <PackageReference Update="Microsoft.Extensions.Options.ConfigurationExtensions" Version="$(ExtensionsVersion)" />
-    <PackageReference Update="Microsoft.Extensions.WebEncoders" Version="$(ExtensionsVersion)" />
+
+    <PackageReference Update="Microsoft.Extensions.WebEncoders" Version="$(AspNetCoreVersion)" />
```
Comment (jakubfijalkowski) [inline Directory.Build.targets:99]: Considering this is `Microsoft.Extensions` package, why are we changing the version to be `AspNetCoreVersion` variable?

**Review by jakubfijalkowski — APPROVED** [review]

File: Directory.Build.targets (hunk 1 above)
Comment (marcin-dardzinski) [reply Directory.Build.targets:99]: Well, the project is in the [aspnetcore](https://github.com/dotnet/aspnetcore) repo and versioning is aligned. 
I can leave a comment or move it to asp net section of this file.

**Comment by Saancreed** [comment]
I'd say let's dump our implementation of `TimeProvider` class and instead let the package be just a static source of `System.TimeProvider` for our purpose that is initialized with `TimeProvider.System` plus maybe a `FixedTimeProvider` that would be a subclass of `System.TimeProvider` for tests, but maybe the latter could exist in a separate package for testing.

File: Directory.Build.targets (hunk 1 above)
Comment (jakubfijalkowski) [reply Directory.Build.targets:99]: Ahh, got it. No need to move, thx.

**Review by jakubfijalkowski — APPROVED** [review]

**Comment by marcin-dardzinski** [comment]
I've noted time provider adjustments in backlog, let's address them separately.

**Pull request merged**